controllers/useraccount/useraccount_controller.go

```diff
@@ -427,24 +428,23 @@ func (r *Reconciler) deleteIdentityAndUser(logger logr.Logger, userAcc *toolchai
 // if the user existed and something wrong happened. If the users don't exist,
 // this func returns `false, nil`
 func (r *Reconciler) deleteUser(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount) (bool, error) {
-	userList := &userv1.UserList{}
-	err := r.Client.List(context.TODO(), userList, listByOwnerLabel(userAcc.Name))
+	userList, err := user.GetUsersByOwnerName(r.Client, userAcc.Name)
```

Comment:
from the code below, it seems like we only treat the first entry in `userList`, so should this `GetUsersByOwnerName` func be renamed `GetUserByOwnerName` instead, and return a single user? and maybe an error if multiple users match?

Reply:
It's a good suggestion but when I tried to change it I realized this function should return `false, nil` if no users are found so then we would have to have a check like `if err.Error() != "no users found" {` so I guess it wouldn't really simplify things much because of this. So maybe we can keep it as is? WDYT?

Reply:
ok, let's keep it as-is for now, then.